auto_update_tests.py

```diff
@@ -2,6 +2,8 @@
 
 import os, sys, subprocess, difflib
 
+os.environ['LD_LIBRARY_PATH'] = 'lib' # find our dynamic libraries
```

Comment:
A tricky thing with static is that it creates a .a file, and that means it only links in methods as needed. But we allow passes to be added by name, and they are registered by globals (like LLVM does commandline param registration). It looks like that wasn't enough, and the passes were not getting linked in.

I would have spent more time to investigate that, but anyhow I think many users will want a dynamic library, e.g. Rust.

If rpath can fix this, that sounds good, I just have only heard about it in this issue :) How would it fix the check.py-compiled programs? (your link was cmake-specific, unless I misunderstood?)